Compilation error on use int variable as array index [$10]

I have a code:

```zep
namespace Project;

class TestClass
{

	public static function doSomething(int param)
	{
		var data, index;

		let data = [];

		let index = param * (int)"123";
		let data[index] = "hello";

		var_dump(data);
	}

}
```

On compilation I got:
```
.../testclass.zep.c:46:34: error: passing 'zval' (aka 'struct _zval_struct') to parameter of incompatible type 'zval *' (aka 'struct _zval_struct *'); take the address with &
        zephir_array_update_zval(&data, index, &_0, PH_COPY | PH_SEPARATE);
                                        ^~~~~
                                        &
./kernel/array.h:50:48: note: passing argument to parameter 'index' here
int zephir_array_update_zval(zval **arr, zval *index, zval **value, int flags);
                                               ^
```

Where testclass.zep.c contains:
```cpp
PHP_METHOD(Project_TestClass, doSomething) {

	zval *param_param = NULL, *data, index, *_0;
	int param;

	ZEPHIR_MM_GROW();
	zephir_fetch_params(1, 1, 0, &param_param);

	param = zephir_get_intval(param_param);


	ZEPHIR_INIT_VAR(data);
	array_init(data);
	ZEPHIR_SINIT_VAR(index);
	ZVAL_LONG(&index, (param * 123));
	ZEPHIR_INIT_VAR(_0);
	ZVAL_STRING(_0, "hello", 1);
	zephir_array_update_zval(&data, index, &_0, PH_COPY | PH_SEPARATE); // 46
	zephir_var_dump(&data TSRMLS_CC);
	ZEPHIR_MM_RESTORE();

}
```

Zephir version 0.5.9a [396cb30]

<bountysource-plugin>
---
There is a **[$10 open bounty](https://www.bountysource.com/issues/6955120-compilation-error-on-use-int-variable-as-array-index?utm_campaign=plugin&utm_content=tracker%2F280146&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F280146&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/Library/Operators/Other/IssetOperator.php
+++ b/Library/Operators/Other/IssetOperator.php
@@ -98,17 +98,17 @@ class IssetOperator extends BaseOperator
 
                     case 'variable':
                         $indexVariable = $compilationContext->symbolTable->getVariableForRead($resolvedExpr->getCode(), $compilationContext, $left['right']);
-                        $indexVariable->setLocalOnly(false);
+                        $indexVariableName = ($indexVariable->isLocalOnly() ? '&' : '') . $indexVariable->getName();
 
                         switch ($indexVariable->getType()) {
 
                             case 'int':
                             case 'long':
-                                return new CompiledExpression('bool', 'zephir_array_isset_long(' . $variable->getName() . ', ' . $indexVariable->getName() . ')', $left['right']);
+                                return new CompiledExpression('bool', 'zephir_array_isset_long(' . $variable->getName() . ', ' . $indexVariableName . ')', $left['right']);
 
                             case 'variable':
                             case 'string':
-                                return new CompiledExpression('bool', 'zephir_array_isset(' . $variable->getName() . ', ' . $indexVariable->getName() . ')', $left['right']);
+                                return new CompiledExpression('bool', 'zephir_array_isset(' . $variable->getName() . ', ' . $indexVariableName . ')', $left['right']);
 
                             default:
                                 throw new CompilerException('[' . $indexVariable->getType() . ']', $expression);

--- a/Library/Statements/Let/ArrayIndex.php
+++ b/Library/Statements/Let/ArrayIndex.php
@@ -190,19 +190,20 @@ class ArrayIndex
 
             case 'variable':
                 $variableIndex = $compilationContext->symbolTable->getVariableForRead($exprIndex->getCode(), $compilationContext, $statement);
+                $variableIndexName = ($variableIndex->isLocalOnly() ? '&' : '') . $variableIndex->getName();
 
                 switch ($variableIndex->getType()) {
                     case 'int':
                     case 'uint':
                     case 'long':
                     case 'ulong':
-                        $codePrinter->output('zephir_array_update_long(&' . $variable . ', ' . $variableIndex->getName() . ', &' . $symbolVariable->getName() . ', ' . $flags . ', "' . Compiler::getShortUserPath($statement['index-expr'][0]['file']) . '", ' . $statement['index-expr'][0]['line'] . ');');
+                        $codePrinter->output('zephir_array_update_long(&' . $variable . ', ' . $variableIndexName . ', &' . $symbolVariable->getName() . ', ' . $flags . ', "' . Compiler::getShortUserPath($statement['index-expr'][0]['file']) . '", ' . $statement['index-expr'][0]['line'] . ');');
                         break;
                     case 'string':
-                        $codePrinter->output('zephir_array_update_zval(&' . $variable . ', ' . $variableIndex->getName() . ', &' . $symbolVariable->getName() . ', ' . $flags . ');');
+                        $codePrinter->output('zephir_array_update_zval(&' . $variable . ', ' . $variableIndexName . ', &' . $symbolVariable->getName() . ', ' . $flags . ');');
                         break;
                     case 'variable':
-                        $codePrinter->output('zephir_array_update_zval(&' . $variable . ', ' . $variableIndex->getName() . ', &' . $symbolVariable->getName() . ', ' . $flags . ');');
+                        $codePrinter->output('zephir_array_update_zval(&' . $variable . ', ' . $variableIndexName . ', &' . $symbolVariable->getName() . ', ' . $flags . ');');
                         break;
                     default:
                         throw new CompilerException("Variable: " . $variableIndex->getType() . " cannot be used as array index", $statement);

